### PR TITLE
ADM-795: [backend]feat: optimize meanToRecovery calculate logic to avoid canceled state effect

### PR DESCRIPTION
**Pull request by sqsq5566**
## Summary

 optimize meanToRecovery calculate logic to avoid canceled state effect

## Before

the result of meanToRecovery is not correct because the logic contains canceled pipeline in it
## After

the result of meanToRecovery is  correct

### CODE REVIEW

File: backend/src/main/java/heartbeat/service/pipeline/buildkite/BuildKiteService.java [hunk 1]
```diff
@@ -318,7 +318,7 @@ public DeployInfo mapToDeployInfo(BuildKiteBuildInfo buildInfo, List<String> ste
 		}
 
 		return new DeployInfo(buildInfo.getPipelineCreateTime(), job.getStartedAt(), job.getFinishedAt(),
-				buildInfo.getCommit(), job.getState(), job.getName());
+				buildInfo.getCommit(), job.getState(), "canceled".equals(buildInfo.getState()), job.getName());
```
Comment (mikeyangyun) [inline backend/src/main/java/heartbeat/service/pipeline/buildkite/BuildKiteService.java:321]: extract `cancelled` to constant, BTW correct spelling is `cancelled`

File: backend/src/main/java/heartbeat/client/dto/pipeline/buildkite/DeployInfo.java [hunk 2]
```diff
@@ -21,6 +21,8 @@ public class DeployInfo {
 
 	private String state;
 
+	private Boolean isPipelineCanceled;
```
Comment (mikeyangyun) [inline backend/src/main/java/heartbeat/client/dto/pipeline/buildkite/DeployInfo.java:24]: maybe boolean is enough ? Or question: what is the benefit using Boolean here

File: backend/src/main/java/heartbeat/service/pipeline/buildkite/BuildKiteService.java (hunk 1 above)
Comment (sqsq5566) [reply backend/src/main/java/heartbeat/service/pipeline/buildkite/BuildKiteService.java:321]: actually canceled is America spell, and cancelled is English spell， they are both correct， and. the state from buildKiteFeign Client is canceled, so we'd better use canceled.
and already extract constant

File: backend/src/main/java/heartbeat/client/dto/pipeline/buildkite/DeployInfo.java (hunk 2 above)
Comment (sqsq5566) [reply backend/src/main/java/heartbeat/client/dto/pipeline/buildkite/DeployInfo.java:24]: yes,boolean is enough, already change it

File: backend/src/main/java/heartbeat/service/pipeline/buildkite/BuildKiteService.java (hunk 1 above)
Comment (mikeyangyun) [reply backend/src/main/java/heartbeat/service/pipeline/buildkite/BuildKiteService.java]: Make sense.

**Review by mikeyangyun — APPROVED** [review]

**Pull request merged**